database/sqlite/sqlite_context.c

```diff
@@ -0,0 +1,597 @@
+    }
+
+
+    rc = sqlite3_bind_text(res, 2, context_data->id, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context id for data deletion");
+        goto failed;
+    }
+
+    rc_stored = execute_insert(res);
+
+    if (rc_stored != SQLITE_DONE)
+        error_report("Failed to delete context %s, rc = %d", context_data->id, rc_stored);
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement where deleting a context, rc = %d", rc);
+
+    return (rc_stored != SQLITE_DONE);
+}
+
+//
+// TESTING FUNCTIONS
+//
+static void dict_ctx_get_label_list_cb(SQL_CLABEL_DATA *label_data_ptr, void *data)
+{
+    (void)data;
+    SQL_CLABEL_DATA *label_data = label_data_ptr;
+
+    info(" LABEL %d %s = %s", label_data->label_source, label_data->label_key, label_data->label_value);
+}
+
+static void dict_ctx_get_dimension_list_cb(SQL_DIMENSION_DATA *dimension_data_ptr, void *data)
+{
+    (void)data;
+
+    SQL_DIMENSION_DATA *dimension_data = dimension_data_ptr;
+
+    char uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower(dimension_data->dim_id, uuid_str);
+
+    info(" Dimension %s = %s", uuid_str, dimension_data->id);
+}
+
+
+static void dict_ctx_get_chart_list_cb(SQL_CHART_DATA *chart_data, void *data)
+{
+    (void)data;
+
+    char uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower(chart_data->chart_id, uuid_str);
+    info("OK GOT %s ID = %s NAME = %s CONTEXT = %s", uuid_str, chart_data->id, chart_data->name, chart_data->context);
+    ctx_get_label_list(&chart_data->chart_id, dict_ctx_get_label_list_cb, NULL);
+    ctx_get_dimension_list(&chart_data->chart_id, dict_ctx_get_dimension_list_cb, NULL);
+}
+
+static void dict_ctx_get_context_list_cb(VERSIONED_CONTEXT_DATA *context_data, void *data)
+{
+    (void)data;
+    info("   Context id = %s "
+         "version = %lu "
+         "title = %s "
+         "chart_type = %s "
+         "units = %s "
+         "priority = %lu "
+         "first time = %lu "
+         "last time = %lu "
+         "deleted = %d"
+         "family = %s",
+         context_data->id,
+         context_data->version,
+         context_data->title,
+         context_data->chart_type,
+         context_data->units,
+         context_data->priority,
+         context_data->first_time_t,
+         context_data->last_time_t,
+         context_data->deleted,
+         context_data->family);
+}
+
+static int localhost_uuid_cb(void *data, int argc, char **argv, char **column)
+{
+    uuid_t *uuid = data;
+    UNUSED(argc);
+    UNUSED(column);
+    uuid_copy(*uuid, * (uuid_t *) argv[0]);
+    return 0;
+}
+
+int ctx_unittest(void)
+{
+    uuid_t host_uuid;
+    uuid_t host_uuid1;
+
+    uuid_generate(host_uuid1);
+
+    char *err_msg;
+
+    sql_init_context_database(1);
+
+    int rc = sqlite3_exec(db_context_meta, "SELECT host_id FROM meta.host WHERE hops = 0;", localhost_uuid_cb, (void *) &host_uuid, &err_msg);
```

Comment:
Instead to pass direct the query to the function, I suggest to create a preprocessor like previous functions.